javascript/javascript1/week4/voiceassitant.js

```diff
@@ -0,0 +1,79 @@
+// Hello my name is Benjamin - Should save the name benjamin. and respond with "nice to meet you Benjamin". What if someone writes this twice?
+// What is my name - should respond with the name of the person. What if the name has not yet been mentioned?
+// Add fishing to my todo - Should respond with "fishing added to your todo". Should add fishing to a list of todos
+// Add singing in the shower to my todo - Should add singing in the shower to a list of todos
+// Remove fishing from my todo - Should respond with "Removed fishing from your todo"
+// What is on my todo? - should respond with the todos. Fx you have 2 todos - fishing and singing in the shower
+// What day is it today? - Should respond with the date in a human readable format. E.g. if today is 30/8/2019 then it should respond with 30. of August 2019
+// Should be able to do simple math. fx what is 3 + 3 should respond with 6. Or what is 4 * 12 should respond with 48
+// Set a timer for 4 minutes - Should respond with "Timer set for 4 minutes". When 4 minutes is up: "Timer done". How do we set a timer in js? Google is your friend here!
+
+let namePerson = "";
+let todoList = [];
+
+function getReply(command) {
+   const partOfCommand = command.split(" ");
+   const action = partOfCommand[0].toLowerCase();
+      
+   if (action === "hello") {
+      let newName = partOfCommand.slice(4).join(" ");
+      if (namePerson === "") {
+         namePerson = newName;
+         return `nice to meet you ${namePerson}`
+      }
+   }
+
+   if (action === "what") {
+      if (partOfCommand[3] === "name") {
+         return `Your name is ${namePerson}`
+      } else if (partOfCommand[4] === "todo?") {
+         if (todoList.length >= 0) {
+            return `You have ${todoList.length} todo - ${todoList}`;
+         }
+      } else if (partOfCommand[4] === "today?") {
+         const currentDay = new Date().toLocaleDateString();
+         return `Today is ${currentDay}`;
+      }
+   }
+
+   if (action === "add") {
+      const todo = partOfCommand.slice(1, -3).join(" ");
```

Comment:
`1, -3` is something called `magic numbers` in programming. It's numbers that solve a problem, but are not very obvious why exactly those numbers are used. It's a good practice to document them through comments - what's the purpose of this number, or to create variables with explanatory names for it.